packages/did/src/Did.chain.ts

```diff
@@ -476,3 +483,82 @@ export function didSignatureToChain(
 
   return { [key.type]: signature } as EncodedSignature
 }
+
+/**
+ * This function fetches the DID document linked to a mnemonic.
+ * @param mnemonic The secret phrase used to fetch data.
+ * @returns  A Full DidDocument.
+ */
+export async function fetchFromMnemonic(mnemonic: string): Promise<DidDocument> {
+  const api = ConfigService.get('api')
+  const {
+    authentication,
+  } = Keys.generateKeypairs(mnemonic,"ed25519")
+  const didUri = getDidUriFromKey(authentication)
+  const encodedDid = await api.call.didApi.query(toChain(didUri))
+  
+  if(encodedDid.isNone){
+    throw new SDKErrors.DidError('No DID is accociated with the provided mnemonic')
+  }
+  else{
+    const { document } = linkedInfoFromChain(encodedDid)
+    return document
+  }
+}
+
+/**
+ * It creates a DID on chain, and returns the mnemonic and DID document
+ * @param submitterAccount - The account that will be used to pay for the transaction.
+ * @returns The mnemonic and the DID document.
+ */
+export async function createDid(
+  submitterAccount: CordKeyringPair,
+  didServiceEndpoint?: DidServiceEndpoint[]
+): Promise<{
+  mnemonic: string
+  document: DidDocument
+}> {
+  const api = ConfigService.get('api')
+
+  const mnemonic = mnemonicGenerate(24)
```

Comment:
Can this do one more change? Can you take mnemonic also as input, and if present, use the same to try create DID with. If not present, then only generate the mnemonic.

Specially in case where a process has to generate DID to start the process, it would be a required step to pass the mnemonic to keep the DID same across restarts.

Reply:
Ok, got it